test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -48,217 +222,493 @@ define(['util/jobLogViewer', 'common/runtime'], (JobLogViewer, Runtime) => {
-                        jobId: jobId,
-                        jobState: {
-                            status: 'running',
-                        },
-                    },
-                    {
-                        channel: {
+            await this.jobLogViewerInstance.start(arg);
+            const btns = this.node.querySelectorAll('div[data-element="header"] button');
+            btns.forEach((btn) => {
+                expect(btn).toHaveClass('disabled');
+            });
+        });
+
+        describe('initial widget state', () => {
+            describe('should be determining the state with no job object', () => {
+                beforeEach(async function () {
+                    await this.jobLogViewerInstance.start({
+                        node: this.node,
+                        jobId: 'no state set',
+                    });
+                });
+                itHasJobStatus();
+            });
+
+            JobsData.validJobs.forEach((jobState) => {
+                describe(`should create a string for status ${jobState.status}`, () => {
+                    beforeEach(async function () {
+                        this.jobState = jobState;
+                        await this.jobLogViewerInstance.start({
+                            node: this.node,
+                            jobId: jobState.job_id,
+                            jobState: jobState,
+                        });
+                    });
+                    itHasJobStatus();
+                });
+            });
+
+            JobsData.invalidJobs.forEach((state) => {
+                describe('should be determining the state with a dodgy job state', () => {
+                    const jobId = 'dodgy_job_state_test';
+                    beforeEach(async function () {
+                        await this.jobLogViewerInstance.start({
+                            node: this.node,
                             jobId: jobId,
-                        },
-                        key: {
-                            type: 'job-status',
-                        },
-                    }
-                );
-                viewer.detach();
-                done();
+                            jobState: state,
+                        });
+                    });
+                    itHasJobStatus();
+                });
             });
-            await viewer.start(arg);
         });
 
-        it('Should start with all buttons disabled', async () => {
-            const viewer = JobLogViewer.make();
-            const jobId = 'testBtnState';
-            const arg = {
-                node: hostNode,
-                jobId: jobId,
-            };
-            await viewer.start(arg);
-            const btns = hostNode.querySelectorAll('div[data-element="header"] button');
-            btns.forEach((btn) => {
-                expect(btn.classList.contains('disabled')).toBeTruthy();
+        describe('initial widget state, history mode on', () => {
```

Comment:
same as the previous set of tests, but this time with history mode on -- i.e. if the job has gone through the queued and running states, info about queue time and run time will be added to the job status lines.